packages/common/src/utils/metricsExplorer.ts

```diff
@@ -77,265 +84,46 @@ export const getFieldIdForDateDimension = (
-                    settings: {
-                        unitOfTime: UnitOfTime.months,
-                        completed: true,
-                    },
-                },
-                {
-                    id: uuidv4(),
-                    target: { fieldId: targetFieldId },
-                    operator: ConditionalOperator.NOT_IN_THE_PAST,
-                    values: [12],
-                    settings: {
-                        unitOfTime: UnitOfTime.months,
-                        completed: true,
-                    },
-                },
-            ];
-        case TimeFrames.YEAR:
-            // 10 years ago - 5 years ago
-            return [
-                {
-                    id: uuidv4(),
-                    target: { fieldId: targetFieldId },
-                    operator: ConditionalOperator.IN_THE_PAST,
-                    values: [10],
-                    settings: {
-                        unitOfTime: UnitOfTime.years,
-                        completed: true,
-                    },
-                },
-                {
-                    id: uuidv4(),
-                    target: { fieldId: targetFieldId },
-                    operator: ConditionalOperator.NOT_IN_THE_PAST,
-                    values: [5],
-                    settings: {
-                        unitOfTime: UnitOfTime.years,
-                        completed: true,
-                    },
-                },
-            ];
-        default:
-            return assertUnimplementedTimeframe(timeInterval);
-    }
-};
-
-// Time grain Year: -> past 5 years (i.e. 5 completed years + this uncompleted year)
-// Time grain Month -> past 12 months (i.e. 12 completed months + this uncompleted month)
-// Time grain Week -> past 12 weeks (i.e. 12 completed weeks + this uncompleted week)
-// Time grain Day -> past 30 days (i.e. 30 completed days + this uncompleted day)
-const getGrainForDateRange = (
+export const oneYearBack = (date: Date) =>
+    dayjs(date)
+        .set('year', dayjs(date).get('year') - 1)
+        .toDate();
+
+export const oneYearForward = (date: Date) =>
+    dayjs(date)
+        .set('year', dayjs(date).get('year') + 1)
+        .toDate();
+
+// TODO: refine the time grain for each time frame
+//   Time grain Year: -> past 5 years (i.e. 5 completed years + this uncompleted year)
+//   Time grain Month -> past 12 months (i.e. 12 completed months + this uncompleted month)
+//   Time grain Week -> past 12 weeks (i.e. 12 completed weeks + this uncompleted week)
+//   Time grain Day -> past 30 days (i.e. 30 completed days + this uncompleted day)
+// above sample is taken from the spec but the implementation is different for practical reasons for visualization
+export const getGrainForDateRange = (
     dateRange: [Date, Date],
 ): ImpelemntedTimeframe => {
     const diff = dateRange[1].getTime() - dateRange[0].getTime();
     const days = diff / (1000 * 60 * 60 * 24);
 
-    if (days <= 31) {
+    if (days <= 31 * 3) {
```

Comment:
is this right?

Reply:
yeah based on the visual appearance of the chart. it just looks sad when you select 3 months filter. we can have a broader discussion on this later 😆